ci/src/main/java/group17/ci/Router.java

```diff
@@ -178,6 +187,58 @@ public void sendCommitStatus(PushPayload payload, CommitStatus status) {
         System.out.println(response);
     }
 
+    /**
+     * Checks a log file for the status of the build and test and returns a string with the status
+     *
+     * @param File the log file containing build and test information
+     * @returns a string specifying log status result.
+     */
+    private static String getStatus(File file) {
+        StringBuilder sb = new StringBuilder();
+        try{
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            int lineCount = 0;
+            //If file contains "BUILD SUCCESS" x2 -> success, "BUILD SUCCESS" x1 -> failed test, else -> build failed
+            int buildSuccesses = 0;
+            String line;
+            while((line = br.readLine()) != null){
+                //Append the date to the status
+                if (lineCount == 0){
+                    sb.append(" [");
+                    sb.append(line.substring(6));
+                    sb.append("] ");
+                }
+                else if(line.contains("BUILD FAILURE"))
+                {
+                    if(buildSuccesses == 1)
+                    {
+                        sb.append("[TESTS FAILED]");
+                        br.close();
+                        return sb.toString();
+                    }
+                    sb.append(" [BUILD FAILURE]");
+                    br.close();
+                    return sb.toString();
+                }
+                else if(line.contains("BUILD SUCCESS"))
+                {
+                    buildSuccesses++;
+                    if (buildSuccesses == 2){
+                        sb.append(" [BUILD SUCCESS]");
+                        br.close();
+                        return sb.toString();
+                    }
+                }
+                lineCount++;
+            }
+            sb.append("[FAILED TO PARSE LOGS]");
+            br.close();
+            return sb.toString();
```

Comment:
Is this dead code or is there a case I'm not considering where it gets executed?